x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/TumblingWindow.java

```diff
@@ -229,6 +411,11 @@ private void baseCriterion(int baseStage, SearchResponse r, ActionListener<Paylo
             }
         } else {
             info = null;
+            // this covers the case where there is only one positive criterion and all the others are missing events
+            if (baseStage == matcher.firstPositiveStage() && base.descending()) {
```

Comment:
I don't see why those two checks cover the scenario described in the comment: "only one positive criterion and all the others are missing events".

Reply:
Yes, there is something wrong (in the checks, not in the scenario). Let me review them.